### ISSUE TITLE
False positive undefined errors for specific files

### ISSUE THREAD

**Issue by bugreportuser** [vimeo/psalm#1417]
Psalm gives many undefined errors for this example, but nothing is undefined, since include.php includes everything. Surprisingly, the example is minimal.

Probably related to https://github.com/vimeo/psalm/issues/1338.

Example:

**include.php**
```php
<?php
include 'func.php';
include 'Base.php';
include 'Child.php';
```

**func.php**
```php
<?php
namespace ns;

function func(): void {}

define('ns\\cons', 0);

cons;
```

**Base.php**
```php
<?php
namespace ns;

func();

cons;

class Base {
    public function __construct() {}
}
```

**Child.php**
```php
<?php
namespace ns;

func();

cons;

class Child extends Base {
    /**
     * @var int
     */
    public $x;

    public function __construct() {
        parent::__construct();
    }
}
```

Example output (using commit 8ec188a):
```
$ psalm include.php
ERROR: UndefinedFunction - Base.php:4:1 - Function ns\func does not exist

ERROR: UndefinedConstant - Base.php:6:1 - Const cons is not defined

ERROR: UndefinedConstant - Child.php:6:1 - Const cons is not defined

INFO: PropertyNotSetInConstructor - Child.php:12:12 - Property ns\Child::$x is not defined in constructor of ns\Child or in any methods called in the constructor

ERROR: UndefinedConstant - func.php:8:1 - Const cons is not defined
```

**Comment by muglug** [comment]
Would you mind creating a repo that reproduces the bug?

**Comment by muglug** [comment]
Nevermind, have reproduced

**Comment by muglug** [comment]
So it's related to constructor property checks. When running nested includes I use a `getRootFilePath` function in `StatementsAnalyzer` to know what the topmost file being analysed is, but that's reset when checking for property initialisation across file boundaries.

**Comment by bugreportuser** [comment]
Thanks for the explanation. Now it makes sense why it was so odd to debug.
